File: VB/SpreadsheetDocServerAPIPart2/CodeExamples/SearchActions.vb
Imports DevExpress.Spreadsheet
Imports System
Imports System.Collections.Generic
Imports System.Drawing

Namespace SpreadsheetDocServerAPIPart2
	Public Module SearchActions
		Public SimpleSearchValueAction As Action(Of Workbook) = AddressOf SimpleSearchValue
		Public AdvancedSearchValueAction As Action(Of Workbook) = AddressOf AdvancedSearchValue

		Private Sub SimpleSearchValue(ByVal workbook As Workbook)
'			#Region "#SimpleSearch"
			workbook.Calculate()
			Dim worksheet As Worksheet = workbook.Worksheets("ExpenseReport")
			workbook.Worksheets.ActiveWorksheet = worksheet

			' Find and highlight cells that contain the word "holiday".
			Dim searchResult As IEnumerable(Of Cell) = worksheet.Search("holiday")
			For Each cell As Cell In searchResult
				cell.Fill.BackgroundColor = Color.LightGreen
			Next cell
'			#End Region ' #SimpleSearch
		End Sub

		Private Sub AdvancedSearchValue(ByVal workbook As Workbook)
'			#Region "#AdvancedSearch"
			workbook.Calculate()
			Dim worksheet As Worksheet = workbook.Worksheets("ExpenseReport")
			workbook.Worksheets.ActiveWorksheet = worksheet

			' Specify the search term.
			Dim searchString As String = Date.Today.ToString("d")

			' Specify search options.
			Dim options As New SearchOptions()
			options.SearchBy = SearchBy.Columns
			options.SearchIn = SearchIn.Values
			options.MatchEntireCellContents = True

			' Find and highlight all cells that contain today's date.
			Dim searchResult As IEnumerable(Of Cell) = worksheet.Search(searchString, options)
			For Each cell As Cell In searchResult
				cell.Fill.BackgroundColor = Color.LightGreen
			Next cell
'			#End Region ' #AdvancedSearch
		End Sub
	End Module
End Namespace

File: CS/SpreadsheetDocServerAPIPart2/CodeExamples/SearchActions.cs
using DevExpress.Spreadsheet;
using System;
using System.Collections.Generic;
using System.Drawing;

namespace SpreadsheetDocServerAPIPart2
{
    public static class SearchActions
    {
        static void SimpleSearchValue(Workbook workbook)
        {
            #region #SimpleSearch
            workbook.Calculate();
            Worksheet worksheet = workbook.Worksheets["ExpenseReport"];
            workbook.Worksheets.ActiveWorksheet = worksheet;

            // Find and highlight cells that contain the word "holiday".
            IEnumerable<Cell> searchResult = worksheet.Search("holiday");
            foreach (Cell cell in searchResult)
                cell.Fill.BackgroundColor = Color.LightGreen;
            #endregion #SimpleSearch
        }

        static void AdvancedSearchValue(Workbook workbook)
        {
            #region #AdvancedSearch
            workbook.Calculate();
            Worksheet worksheet = workbook.Worksheets["ExpenseReport"];
            workbook.Worksheets.ActiveWorksheet = worksheet;

            // Specify the search term.
            string searchString = DateTime.Today.ToString("d");

            // Specify search options.
            SearchOptions options = new SearchOptions();
            options.SearchBy = SearchBy.Columns;
            options.SearchIn = SearchIn.Values;
            options.MatchEntireCellContents = true;

            // Find and highlight all cells that contain today's date.
            IEnumerable<Cell> searchResult = worksheet.Search(searchString, options);
            foreach (Cell cell in searchResult)
                cell.Fill.BackgroundColor = Color.LightGreen;
            #endregion #AdvancedSearch
        }
    }
}
